### PR TITLE
Stage file items

### PR DESCRIPTION
**Pull request by jo-pol**
Work in progess

### CODE REVIEW

File: src/main/scala/nl/knaw/dans/easy/stage/EasyStageFileItem.scala [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  def findParent(implicit s: FileItemSettings): Try[String] = {
+    val parentFile = s.filePath.getParentFile
+    if (parentFile.toString=="") Success(s.datasetId)
+    else EasyFilesAndFolders.getFolder(parentFile, s.datasetId)
+  }
+
+  private def createFileSdo(file: File, ownerId: String)(implicit s: FileItemSettings): Try[Unit] = {
+    log.debug(s"Creating file SDO for $file")
+    val relativePath = file.getPath.replaceFirst(s.bagitDir.getPath, "").substring(1)
+    val bagLocation = s"${??? /*s.bagStorageLocation*/}/$relativePath"
```
Comment (jo-pol) [inline src/main/scala/nl/knaw/dans/easy/stage/EasyStageFileItem.scala:54]: @janvanmansum @GeorgiKhomeriki 

The implicit settings cause a few problems, the old and new main objects need different settings, but down the line a single one is expected. I only have a file for the new command, the metadata does not come from the bag but from the command line or CSV-file.

A super class for the Settings objects and some refactoring (a.o. variants for JSON.createFileCfg) seems to fix at least some of the issues. I'm still struggling with bagitDir and bagStorageLocation above, bagitDir is also used via getSDODir below. There are still other holes to fix.

File: src/main/assembly/dist/bin/easy-stage-file-item [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+BINPATH=`readlink -f $0`
+if [ $? -ne 0 ] ; then
```
Comment (janvanmansum) [inline src/main/assembly/dist/bin/easy-stage-file-item:4]: What does this do?

File: src/main/assembly/dist/cfg/application.properties [hunk 3]
```diff
@@ -3,3 +3,5 @@ storage-base-url={{ easy_stage_dataset_storage_baseurl }}
 fcrepo-service-url={{ easy_stage_dataset_fcrepo_service_url }}
 fcrepo-user={{ easy_stage_dataset_fcrepo_user }}
 fcrepo-password={{ easy_stage_dataset_fcrepo_password }}
+db-connection-url={{ easy_read_fs_rdb_db_connection_url }}
```
Comment (janvanmansum) [inline src/main/assembly/dist/cfg/application.properties:6]: Use placeholder that starts with `easy_stage_dataset_`

File: src/main/scala/nl/knaw/dans/easy/stage/FileItemConf.scala [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import org.rogach.scallop.{ScallopConf, ValueConverter, singleArgConverter}
+import org.slf4j.LoggerFactory
+
+class FileItemConf(args: Seq[String]) extends ScallopConf(args) {
+  val log = LoggerFactory.getLogger(getClass)
+
+  printedName = "easy-stage-fiel-item"
+  version(s"$printedName v${Version()}")
+  banner(s"""
+            |Stage a file item in EASY-BagIt format for ingest into a datasaet in an EASY Fedora Commons 3.x Repository.
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/stage/FileItemConf.scala:16]: Remove "in EASY-BagIt format".

File: src/main/scala/nl/knaw/dans/easy/stage/FileItemConf.scala [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import java.util.UUID
+
+import org.joda.time.DateTime
+import org.rogach.scallop.{ScallopConf, ValueConverter, singleArgConverter}
+import org.slf4j.LoggerFactory
+
+class FileItemConf(args: Seq[String]) extends ScallopConf(args) {
+  val log = LoggerFactory.getLogger(getClass)
+
+  printedName = "easy-stage-fiel-item"
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/stage/FileItemConf.scala:13]: fiel -> file

File: src/main/scala/nl/knaw/dans/easy/stage/FileItemConf.scala [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+      throw new IllegalArgumentException()
+    }
+    new File(f)
+  })
+
+  val filePath = opt[File](
+    name = "file-path", short = 'p',
+    descr = "the path that the file should get in the dataset"
+    )(mayNotExist)
+  val identifier = opt[UUID](
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/stage/FileItemConf.scala:45]: A UUID?

**Comment by janvanmansum** [comment]
Please, make sure that your code at least builds even if it is still in draft state. 

Otherwise a code review is _either_ bound to be rather superficial _or_ the reviewer must first resolve the build errors. The last option means the reviewer is wasting time on something that is going to be done again before the final pull request.

File: src/main/assembly/dist/bin/easy-stage-file-item (hunk 2 above)
Comment (jo-pol) [inline src/main/assembly/dist/bin/easy-stage-file-item:4]: the then-branch is a fall-back for the mac

**Comment by jo-pol** [comment]
As the last commit message says, I even checked that ingest-flow still compiled without updates and stage-dataset still executed with the sample-bag. So there must be either some last-minute mistake or a difference with your environment.

PS: I see now, README tests fail

File: src/main/scala/nl/knaw/dans/easy/stage/FileItemConf.scala (hunk 6 above)
Comment (jo-pol) [inline src/main/scala/nl/knaw/dans/easy/stage/FileItemConf.scala:45]: Inspired by the sample value found in example-bag/metadata/files.xml

**Comment by jo-pol** [comment]
Can't push any more to this branch as I rebased on the tip of DANS-KNAW:master

**Pull request closed without merging**